### PR TITLE
added fees (feature request #69)

### LINKED ISSUE #69: [Feature Request] Extra Fees for Orders
Thank you for open wc-smooth-generator to the public. We've integrated this to test our extensions for Woocommerce better and make them more reliable.

**Is your feature request related to a problem? Please describe.**
Currently created orders are containing only basic data. They dont use the extra fees option inside woo. A lot plugins use this functionality to add further fees to the order.

**Describe the solution you'd like**
It would be very nice to add some fees to the order. I would add some example fee amounts : $10.00 (fixed) , $100.00 (fixed), $1000.00 (fixed),  10% (percentage), 33% (percentage). I would add it to every (lets say) every 5. order.

We understand that this repo is there to easily gather some testdata and not have a complete set of each use-case that would be possible in woo. Still we believe this could be interesting for a wider audience.

### PR DESCRIPTION
**Pull request by michi-dev**
### All Submissions:

* [x] Have you followed the [Contributing guidelines](https://github.com/woocommerce/wc-smooth-generator/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/wc-smooth-generator/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Closes #69.

"It would be very nice to add some fees to the order. I would add some example fee amounts : $10.00 (fixed) , $100.00 (fixed), $1000.00 (fixed), 10% (percentage), 33% (percentage). I would add it to every (lets say) every 5. order."

- every 5. order (randomized function so not particular ever 5.) gets a fee
- the fee is a float between 0.05 and 100 (used https://github.com/fzaninotto/Faker )
- I just used absolute values -> because you only can insert relative percentages via the graphical interface. After that it will be set as absolute values

### How to test the changes in this Pull Request:

1. use the generall command `wp wc generate orders <amount>` at least with amount 5 
2. remeber that its random -> so it could depend if its every 5. order

### Other information:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [X] Have you successfully run tests with your changes locally?

### Changelog entry
add fees to orders randomly (every 5. order, fees from 0.05 - 100 in specific currency)
> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by masteradhoc** [comment]
@rrennick any chance someone of your team can take a look at it anytime next?

**Comment by rrennick** [comment]
@masteradhoc Thanks for the ping. I'm working on a non-Woo project ATM. @ObliviousHarmony Could you follow up here?

File: includes/Generator/Order.php [hunk 1]
```diff
@@ -58,6 +58,28 @@ public static function generate( $save = true, $assoc_args = array() ) {
 		$order->set_shipping_postcode( $customer->get_shipping_postcode() );
 		$order->set_shipping_state( $customer->get_shipping_state() );
 		$order->set_shipping_country( $customer->get_shipping_country() );
+
+		if ( self::$faker->randomFloat( 0, 1, 5 ) == 1 ) {
```
Comment (Luc45) [inline includes/Generator/Order.php]: Minor suggestion for readability
```suggestion
		// 20% chance
		if ( rand( 0, 100 ) <= 20 ) {
```

**Review by Luc45 — APPROVED** [review]
I have tested this locally and it works great. Thanks for the contribution!

(I will forward this internally to see if we can get it merged)

File: includes/Generator/Order.php (hunk 1 above)
Comment (Luc45) [reply includes/Generator/Order.php]: @michi-dev would you mind committing this suggestion?

Comment (michi-dev) [reply includes/Generator/Order.php]: @Luc45 Yes no problem :)

**Comment by michi-dev** [comment]
changes have been done @Luc45

**Comment by masteradhoc** [comment]
@Luc45 any update on these PR's?

**Comment by samueljseay** [comment]
I see no reason why we can't merge this. Thanks for this @michi-dev and apologies this has been sitting around so long.

**Pull request merged**